openfisca_web_api_preview/scripts/serve.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+
+import os
+import sys
+import argparse
+from gunicorn.app.base import BaseApplication
+from gunicorn.six import iteritems
+
+from openfisca_core.scripts import add_tax_benefit_system_arguments, detect_country_package
+from ..app import create_app
+
+
+DEFAULT_PORT = '5000'
+BASE_CONF = {
+    'bind': '%s:%s' % ('127.0.0.1', DEFAULT_PORT),
+    'workers': '3',
+}
+
+
+def new_configured_app():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-p', '--port', action = 'store', default = DEFAULT_PORT, help = "port to serve on", type = int)
+    parser.add_argument('-t', '--tracker_url', action = 'store', help = "tracking service url", type = str)
+    parser.add_argument('-s', '--tracker_idsite', action = 'store', help = "tracking service id site", type = int)
+    parser = add_tax_benefit_system_arguments(parser)
+    args = parser.parse_args()
+
+    app = create_app(args.country_package, args.extensions, args.tracker_url, args.tracker_idsite)
+    return app
+
+
+class StandaloneApplication(BaseApplication):
+
+    def __init__(self, app, options=None):
+        self.options = options or {}
+        self.application = app
+        super(StandaloneApplication, self).__init__()
+
+    def load_config(self):
+        config = dict([(key, value) for key, value in iteritems(self.options)
+                       if key in self.cfg.settings and value is not None])
+        for key, value in iteritems(config):
+            self.cfg.set(key.lower(), value)
```

Comment:
This function seems very complicated for what it does. What about:

```
    def load_config(self):
        for key, value in self.options.iteritems():
            if key in self.cfg.settings and value is not None:
                self.cfg.set(key.lower(), value)
```

But why checking `if key in self.cfg.settings and value is not None` ?
- if the user somehow pass a wrong config item, is it really a good idea to filter out the wrong config item, and not throw any error ?
- In which case `value` can be `None`, in the current `implem` ?

Reply:
Function updated.
No information is logged if `key not in self.cfg.settings` as it doesn't accept `port` or any other python variable that might be useful for our users.

But an information is now logged for `value is None` and the execution isn't interrupted. If no value is given in the configuration file, an error is thrown as the python configuration file won't be valid.